### PR TITLE
Update project to use apple/swift-crypto

### PR DESCRIPTION
**Pull request by edwellbrook**
- Updates project to use [`apple/swift-crypto`](https://github.com/apple/swift-crypto) library
- Includes some other changes preparing new cipher suites
- Includes some helpers for use as a library

---
~We should complete [this issue](https://github.com/samueltangz/swift-noise-protocol/issues/18) before merging to ensure new updates have tests run against them.~

### CODE REVIEW

File: Sources/SwiftNoise/Components/Curve.swift [hunk 1]
```diff
@@ -26,35 +25,49 @@ public protocol Curve {
...
 }
 
-public class C25519: Curve {
-  public init() {}
+func randomKey(_ count: Int) -> SecretKey {
+  let bytes = (0..<count).map({ _ in UInt8.random(in: 0...UInt8.max) })
+  return Data(bytes)
+}
 
-  func normalize(secretKey: SecretKey) -> SecretKey {
```
Comment (edwellbrook) [inline Sources/SwiftNoise/Components/Curve.swift:32]: What was the reason for having this @samueltangz? Couldn't find anything about normalising keys in the spec 🤔 

Maybe I'm missing something?

Comment (samueltangz) [reply Sources/SwiftNoise/Components/Curve.swift:32]: It was suggested by D. J. Bernstein [here](https://cr.yp.to/ecdh.html), inside the paragraph _Computing secret keys_. Its existence was to serve as a hack for computing public keys. We were using [christophhagen/CryptoKit25519](https://github.com/christophhagen/CryptoKit25519) and they didn't provide this as a functionality. Related issue: https://github.com/christophhagen/CryptoKit25519/issues/1.

Comment (edwellbrook) [reply Sources/SwiftNoise/Components/Curve.swift:32]: Thank you! I believe this implementation using apple/crypto works as expected and doesn't need the `normalize` function.

Added a test to confirm the public and private keys match the expected values from here: https://github.com/tgalal/dissononce/blob/1a5b468a4b8cc4547980e89f69ac9dd1ff2a0095/tests/vectors/snow.txt#L256

**Comment by edwellbrook** [comment]
Ready for you to review now. Let me know if you have any issues/suggestions.

Thanks!

**Comment by samueltangz** [comment]
Thanks in advance for the PR! I may need some more time to read through the code for the security details. Hope this wouldn't slow you down a lot.

**Review by samueltangz — APPROVED** [review]

**Pull request merged**